forester/src/slot_tracker.rs

```diff
@@ -0,0 +1,108 @@
+use light_test_utils::rpc::rpc_connection::RpcConnection;
+use log::{debug, warn};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use tokio::sync::Mutex;
+use tokio::time::{interval, sleep};
+
+#[derive(Debug)]
+pub struct SlotTracker {
+    last_known_slot: u64,
+    last_update_time: Instant,
+    update_interval: Duration,
+    slot_duration: Duration,
+}
+
+impl SlotTracker {
+    pub async fn new<R: RpcConnection>(rpc: &mut R) -> crate::Result<Self> {
+        let slot = rpc.get_slot().await?;
+        Ok(Self {
+            last_known_slot: slot,
+            last_update_time: Instant::now(),
+            update_interval: Duration::from_secs(30),
+            slot_duration: Duration::from_millis(400),
```

Comment:
would it make sense to make these values configurable?

Reply:
We should check whether slot time can drift or only block time drifts.
Additionally, we could check whether we can fetch average slot time from Solana rpc at startup if not its likely a hardcoded value and we should lookup the constant.